### PR TITLE
wgengine: re-set DNS config on Linux after a major link change

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #2458 (maybe fixes it)

### CODE REVIEW

File: wgengine/userspace.go [hunk 1]
```diff
@@ -1076,6 +1078,23 @@ func (e *userspaceEngine) linkChange(changed bool, cur *interfaces.State) {
 	health.SetAnyInterfaceUp(up)
 	e.magicConn.SetNetworkUp(up)
 
+	// Hacky workaround for Linux DNS issue 2458: on
+	// suspend/resumse or whenever NetworkManager is started, it
```
Comment (DentonGentry) [inline wgengine/userspace.go:1082]: resumse

File: wgengine/userspace.go [hunk 2]
```diff
@@ -1076,6 +1078,23 @@ func (e *userspaceEngine) linkChange(changed bool, cur *interfaces.State) {
...
+	// nukes all systemd-resolved configs. So reapply our DNS
+	// config on major link change.
+	if runtime.GOOS == "linux" && changed {
```
Comment (DentonGentry) [inline wgengine/userspace.go:1085]: Presumably we only need to do this for the systemd-resolved case. I guess it has the downside of rewriting /etc/resolv.conf more often in the Direct case, if users have written scripts to make local modifications after Tailscale comes up. However its also likely to fix other cases of broken DNS, just by rewriting more often.

**Review by DentonGentry — APPROVED** [review]

**Review by danderson — APPROVED** [review]

File: wgengine/userspace.go (hunk 2 above)
Comment (crawshaw) [reply wgengine/userspace.go:1085]: Elsewhere we have an issue suggesting we fsnotify resolv.conf and rewrite it on change, so we always win. I think that’s generally the right attitude: if we think we own it, own it.

**Review by crawshaw — APPROVED** [review]

**Comment by frioux** [comment]
I installed locally built versions (see this:)

```bash
$ tailscale version 
date.20210603
  go version: go1.16.5
$ tailscaled -version
date.20210603
  go version: go1.16.5
```

But when I try to start it I get this:

```bash
$ sudo tailscale up
2021/07/26 08:38:57 GotNotify: Version mismatch! frontend="date.20210603" backend="1.11.137-ta5fb8e073-gfad17047b"
backend error: GotNotify: Version mismatch! frontend="date.20210603" backend="1.11.137-ta5fb8e073-gfad17047b"
```

**Comment by DentonGentry** [comment]
That means tailscaled is still the release version, while the tailscale CLI command is the one you built.

If you want to permanently switch to the version you built:
1. go build cmd/tailscaled
1. sudo systemctl stop tailscaled
1. sudo cp tailscaled /usr/sbin/tailscaled
1. sudo systemctl start tailscaled

**Comment by frioux** [comment]
Ah I didn't realize tailscale down was different from stopping the service.  Thanks.  This fixed it!

**Pull request merged**